pkg/kubelet/config/common.go

```diff
@@ -69,6 +69,10 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
 		klog.V(5).InfoS("Generated UID", "pod", klog.KObj(pod), "podUID", pod.UID, "source", source)
 	}
 
+	if pod.Name == "" {
+		return fmt.Errorf("name is needed for the pod")
```

Comment:
I'd suggest to move this code to the top of the function to avoid unnecessary UID generation for the pod without a name.